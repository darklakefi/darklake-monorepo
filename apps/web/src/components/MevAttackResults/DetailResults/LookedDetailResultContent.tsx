import Tabs from "@/components/Tabs";
import SummaryTab from "./SummaryTab";
import Button from "@/components/Button";

const TAB_NAMES = [
  {
    text: "[SUMMARY]",
  },
  {
    text: "[ATTACKS]",
  },
];

interface LookedDetailResultContentProps {
  onConnect: () => void;
  connectWithTwitterDisabled: boolean;
}

export default function LookedDetailResultContent(props: LookedDetailResultContentProps) {
  const handleConnect = () => {
    if (props.connectWithTwitterDisabled) {
      return;
    }
    props?.onConnect();
  };

  return (
    <div className="relative">
      <div className="">
        <Tabs tabs={TAB_NAMES.map((tab) => tab.text)} wrapperClassName="py-[16px] gap-[24px]" activeTab={0} />
        <div>
          <SummaryTab.Blurred />
        </div>
      </div>

      <div className="absolute inset-0 flex items-center justify-center bg-[#010F06F2] bg-opacity-95">
        <div className="flex flex-col gap-[40px]">
          <div className="flex flex-col gap-[20px] text-body-2 text-center">
            <div> ******************************** </div>
            <div className="uppercase"> * ACCESS DENIED * </div>
            <div> ******************************** </div>
          </div>
          <Button intent="primary-light" onClick={handleConnect} disabled={props.connectWithTwitterDisabled}>
            Connect <i className="hn hn-x text-brand-30 mx-2" /> to access complete case file
          </Button>
        </div>
      </div>
    </div>
  );
}
