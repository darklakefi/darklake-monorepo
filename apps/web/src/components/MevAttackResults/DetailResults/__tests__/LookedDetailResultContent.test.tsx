import { act, render, RenderResult } from "@testing-library/react";
import LookedDetailResultContent from "../LookedDetailResultContent";

describe("<LookedDetailResultContent />", () => {
  it("displays content successfully", async () => {
    let baseElement: RenderResult["baseElement"] | undefined;

    await act(async () => {
      ({ baseElement } = render(<LookedDetailResultContent onConnect={() => {}} connectWithTwitterDisabled={false} />));
    });

    expect(baseElement).toMatchSnapshot();
  });
});
