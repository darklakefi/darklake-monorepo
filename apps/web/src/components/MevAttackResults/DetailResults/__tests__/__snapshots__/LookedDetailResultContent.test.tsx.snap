// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LookedDetailResultContent /> displays content successfully 1`] = `
<body>
  <div>
    <div
      class="relative"
    >
      <div
        class=""
      >
        <div
          class="flex flex-row items-center py-[16px] gap-[24px]"
        >
          <button
            class="text-lg p-[8px] cursor-pointer select-none bg-brand-60 text-brand-20"
            type="button"
          >
            [SUMMARY]
          </button>
          <button
            class="text-lg p-[8px] cursor-pointer select-none text-brand-30"
            type="button"
          >
            [ATTACKS]
          </button>
        </div>
        <div>
          <div
            class="flex flex-col gap-4"
          >
            <div
              class="flex flex-row gap-4"
            >
              <div
                class="flex-1"
              >
                <div
                  class="flex flex-col bg-brand-60 gap-[16px] p-[16px] h-full"
                >
                  <div
                    class="text-body-2 text-brand-30"
                  >
                    Total Extracted
                  </div>
                  <div
                    class="flex flex-col gap-1 py-0.5"
                  >
                    <div
                      class="text-body text-brand-20"
                    >
                      ██.██ SOL
                    </div>
                    <div
                      class="text-body-2 text-brand-30"
                    >
                      ████.██ USDC
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="flex-1 max-md:hidden"
              >
                <div
                  class="flex flex-col bg-brand-60 gap-[16px] p-[16px] h-full"
                >
                  <div
                    class="text-body-2 text-brand-30"
                  >
                    Confirmed Attack
                  </div>
                  <div
                    class="flex flex-col gap-1 py-0.5"
                  >
                    <div
                      class="text-body text-brand-20"
                    >
                      ██
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex flex-row flex-wrap gap-4"
            >
              <div
                class="flex-1"
              >
                <div
                  class="flex flex-col bg-brand-60 gap-[16px] p-[16px] h-full"
                >
                  <div
                    class="flex flex-col"
                  >
                    <div
                      class="text-body-2 text-brand-20"
                    >
                      #
                      1
                       LARGEST EXTRACTION EVENT
                    </div>
                    <div
                      class="text-body-2 text-brand-30"
                    >
                      TOKEN: █████
                    </div>
                    <div
                      class="text-body-2 text-brand-30"
                    >
                      ████-██-██ ██:██ UTC
                    </div>
                  </div>
                  <div
                    class="flex flex-col gap-[12px] py-[16px] border-t border-b border-brand-40"
                  >
                    <div
                      class="text-body"
                    >
                      ██.██ SOL LOST
                    </div>
                    <div
                      class="text-body-2"
                    >
                      ░░░░░░░░░░░░░░░░░░░░░░░░░░
                    </div>
                    <div
                      class="flex flex-col"
                    >
                      <div
                        class="text-body-2 text-brand-20"
                      >
                        ██% EXTRACTED
                      </div>
                      <div
                        class="text-body-2 text-brand-30"
                      >
                        FROM A ██.██ SOL TRANSACTION
                      </div>
                    </div>
                  </div>
                  <button
                    class="text-link text-brand-30 w-fit"
                  >
                    View attack breakdown
                  </button>
                </div>
              </div>
              <div
                class="flex-1 max-md:hidden"
              >
                <div
                  class="flex flex-col bg-brand-60 gap-[16px] p-[16px] h-full"
                >
                  <div
                    class="flex flex-col"
                  >
                    <div
                      class="text-body-2 text-brand-20"
                    >
                      #
                      2
                       LARGEST EXTRACTION EVENT
                    </div>
                    <div
                      class="text-body-2 text-brand-30"
                    >
                      TOKEN: █████
                    </div>
                    <div
                      class="text-body-2 text-brand-30"
                    >
                      ████-██-██ ██:██ UTC
                    </div>
                  </div>
                  <div
                    class="flex flex-col gap-[12px] py-[16px] border-t border-b border-brand-40"
                  >
                    <div
                      class="text-body"
                    >
                      ██.██ SOL LOST
                    </div>
                    <div
                      class="text-body-2"
                    >
                      ░░░░░░░░░░░░░░░░░░░░░░░░░░
                    </div>
                    <div
                      class="flex flex-col"
                    >
                      <div
                        class="text-body-2 text-brand-20"
                      >
                        ██% EXTRACTED
                      </div>
                      <div
                        class="text-body-2 text-brand-30"
                      >
                        FROM A ██.██ SOL TRANSACTION
                      </div>
                    </div>
                  </div>
                  <button
                    class="text-link text-brand-30 w-fit"
                  >
                    View attack breakdown
                  </button>
                </div>
              </div>
              <div
                class="flex-1 max-md:hidden"
              >
                <div
                  class="flex flex-col bg-brand-60 gap-[16px] p-[16px] h-full"
                >
                  <div
                    class="flex flex-col"
                  >
                    <div
                      class="text-body-2 text-brand-20"
                    >
                      #
                      3
                       LARGEST EXTRACTION EVENT
                    </div>
                    <div
                      class="text-body-2 text-brand-30"
                    >
                      TOKEN: █████
                    </div>
                    <div
                      class="text-body-2 text-brand-30"
                    >
                      ████-██-██ ██:██ UTC
                    </div>
                  </div>
                  <div
                    class="flex flex-col gap-[12px] py-[16px] border-t border-b border-brand-40"
                  >
                    <div
                      class="text-body"
                    >
                      ██.██ SOL LOST
                    </div>
                    <div
                      class="text-body-2"
                    >
                      ░░░░░░░░░░░░░░░░░░░░░░░░░░
                    </div>
                    <div
                      class="flex flex-col"
                    >
                      <div
                        class="text-body-2 text-brand-20"
                      >
                        ██% EXTRACTED
                      </div>
                      <div
                        class="text-body-2 text-brand-30"
                      >
                        FROM A ██.██ SOL TRANSACTION
                      </div>
                    </div>
                  </div>
                  <button
                    class="text-link text-brand-30 w-fit"
                  >
                    View attack breakdown
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="absolute inset-0 flex items-center justify-center bg-[#010F06F2] bg-opacity-95"
      >
        <div
          class="flex flex-col gap-[40px]"
        >
          <div
            class="flex flex-col gap-[20px] text-body-2 text-center"
          >
            <div>
               ******************************** 
            </div>
            <div
              class="uppercase"
            >
               * ACCESS DENIED * 
            </div>
            <div>
               ******************************** 
            </div>
          </div>
          <button
            class="uppercase font-secondary text-lg leading-6 button bg-brand-10 text-brand-70 px-[12px] py-[5px] hover:bg-brand-20 focus:outline-none active:ring-[1px] active:bg-brand-10 active:ring-brand-10 active:ring-offset-[3px] active:ring-offset-black disabled:active:bg-brand-10"
          >
            <span
              class="flex flex-row justify-center items-center"
            >
              Connect 
              <i
                class="hn hn-x text-brand-30 mx-2"
              />
               to access complete case file
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
